core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/edit/PersonEditViewModel.kt

```diff
@@ -288,9 +294,20 @@ class PersonEditViewModel(
     }
 
     fun onEntityChanged(entity: Person?) {
+        val filteredEntity = entity?.let { currentEntity ->
```

Comment:
The logic you have here is correct; that we should only filter the username if it was changed. I think this would be clearer if it starts with the if statement, and I think that can go into the existing update e.g.

```
_uiState.update { prev -> 
    prev.copy(
        person = if(entity?.username != _uiState.value.person?.username) {
            entity?.shallowCopy { 
                username = filterUsernameUseCase(
                     username = entity.username ?: "",
                     invalidCharReplacement = ""
               )
            } 
        }else {
             entity
        }  
    )
}
```